plaso/parsers/spotlight_storedb.py

```diff
@@ -896,19 +899,50 @@ def _ReadRecordPage(self, file_object, file_offset):
     page_header, bytes_read = self._ReadPropertyPageHeader(
         file_object, file_offset)
 
-    if page_header.property_table_type != 0x00000009:
+    if page_header.property_table_type not in (0x00000009, 0x00001009):
       raise errors.ParseError(
           'Unsupported property table type: 0x{0:08x}'.format(
               page_header.property_table_type))
 
     page_data = file_object.read(page_header.page_size - bytes_read)
 
+    file_offset += bytes_read
+
     if page_header.uncompressed_page_size > 0:
-      # TODO: add support for other compression types.
-      if page_data[0] != 0x78:
-        raise errors.ParseError('Unsupported compression type')
+      compressed_page_data = page_data
+
+      if (page_header.property_table_type == 0x00000009 and
+          compressed_page_data[0] == 0x78):
+        page_data = zlib.decompress(compressed_page_data)
+
+      elif (page_header.property_table_type == 0x00001009 and
+            compressed_page_data[0:4] == b'bv41'):
+        data_type_map = self._GetDataTypeMap(
+            'spotlight_store_db_lz4_block_header')
 
-      page_data = zlib.decompress(page_data)
+        try:
+          lz4_block_header = data_type_map.MapByteStream(compressed_page_data)
+        except dtfabric_errors.MappingError as exception:
+          raise errors.ParseError((
+              'Unable to map LZ4 bock header at offset: 0x{0:08x} with error: '
+              '{1!s}').format(file_offset, exception))
+
+        end_of_compressed_data_offset = (
+            12 + lz4_block_header.compressed_data_size)
+
+        page_data = lz4.block.decompress(
+            compressed_page_data[12:end_of_compressed_data_offset],
+            uncompressed_size=lz4_block_header.uncompressed_data_size)
+
+        end_of_compressed_data_identifier = compressed_page_data[
+            end_of_compressed_data_offset:end_of_compressed_data_offset + 4]
+
+        if end_of_compressed_data_identifier != b'bv4$':
+          raise errors.ParseError(
+              'Unsupported LZ4 end of compressed data marker')
```

Comment:
Please include the value of end_of_compressed_data_identifier in the error message

Reply:
4 bytes of random data is not going to tell much. I'll add the offset instead.